pipeline-steps/keyvault-read.yaml

```diff
@@ -5,49 +5,20 @@ parameters:
 
 steps:
 
-- task: AzureCLI@2
-  displayName: Determine Key Vault and Storage names
-  name: SetControlVars
-  inputs:
-    azureSubscription: $(serviceConnection)
-    scriptType: 'bash'
-    scriptLocation: 'inlineScript'
-    inlineScript: |
-      az account set -s $(serviceConnection)
-      SUB_ID=$(az account show --query id -o tsv)
-      TENANT_ID=$(az account show --query tenantId -o tsv)
-
-      echo $SUB_ID
-      echo "##vso[task.setvariable variable=ARM_SUBSCRIPTION_ID]$SUB_ID"
-      echo $TENANT_ID
-      echo "##vso[task.setvariable variable=ARM_TENANT_ID]$TENANT_ID"
-
-      FIRST=$(echo $SUB_ID | cut -c 1-8)
-      SECOND=$(echo $SUB_ID | cut -c 25-36)
-      JOINT=c${FIRST}${SECOND}
-      
-      KEY_VAULT=${JOINT}kv
-      STORAGE_ACCOUNT=${JOINT}sa
-
-      echo KeyVault: $KEY_VAULT
-      echo "##vso[task.setvariable variable=controlKeyVault]$KEY_VAULT"
-      echo Storage account: $STORAGE_ACCOUNT
-      echo "##vso[task.setvariable variable=controlStorageAccount]$STORAGE_ACCOUNT"
-
-      lowerCaseEnv=$(echo "${{ parameters.environment }}" | tr "[:upper:]" "[:lower:]")
-      echo "##vso[task.setvariable variable=env]$lowerCaseEnv"
-      echo Lower case env: $lowerCaseEnv
-
-
-- task: AzureKeyVault@1
-  displayName: 'Get Secrets from KeyVault'
-  inputs:
-    azureSubscription: ${{ parameters.serviceConnection }}
-    keyVaultName: $(controlKeyVault)
-    secretsFilter: 'sp-application-id,sp-token'
-
-- script: |
-    echo '##vso[task.setvariable variable=ARM_CLIENT_ID]$(sp-application-id)'
-    echo '##vso[task.setvariable variable=ARM_CLIENT_SECRET]$(sp-token)'
-  displayName: 'Export Keyvault Variables'
-  name: SetSubscriptionSP
+  - template: steps/keyvault-read.yaml@cnp-azuredevops-libraries
+    parameters:
+      serviceConnection: ${{ parameters.serviceConnection }}
+      environment: ${{ parameters.environment }}
+
+  - task: AzureKeyVault@1
+    displayName: 'Get Secrets from KeyVault'
+    inputs:
+      azureSubscription: ${{ parameters.serviceConnection }}
+      keyVaultName: $(controlKeyVault)
+      secretsFilter: 'sp-application-id,sp-token'
+
+  - script: |
+      echo '##vso[task.setvariable variable=ARM_CLIENT_ID]$(sp-application-id)'
+      echo '##vso[task.setvariable variable=ARM_CLIENT_SECRET]$(sp-token)'
```

Comment:
future enhancement likely, I'm pretty sure these aren't used at all, could be cleaned up separately

Reply:
I dont think they are used at all - @vijay-tenoy can you run a plan with these commented out?

Reply:
Sure, will cleanup and report back here

Reply:
Removed  keyvault-read.yaml entirely and directly used `steps/keyvault-read.yaml@cnp-azuredevops-libraries` in templates that needed to read from keyvault.

Reply:
cool that one looks more sensible:
https://github.com/hmcts/cnp-azuredevops-libraries/blob/master/steps/keyvault-read.yaml

it just sets useful vars like sub id, tenant id, vault name and storage account